### PR TITLE
Convert to core22, update ENV, dependencies, snapcraft changes.

### PR DESCRIPTION
**Pull request by ScarlettGatelyMoore**
Convert to core22 and everything that comes with it. This builds and runs on my system. Needs testing by a gifted cad developer :)

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -114,27 +124,29 @@ apps:
 package-repositories:
   - type: apt
     ppa: elmer-csc-ubuntu/elmer-csc-ppa
-  - type: apt
-    ppa: openscad/releases
```
Comment (luzpaz) [inline snap/snapcraft.yaml:118]: Why is openscad deleted ?

Comment (ScarlettGatelyMoore) [reply snap/snapcraft.yaml:118]: The last release in that PPA is for focal. core22 is Jammy, openscad is in the official repositories for jammy so you will see further down I added it to build/stage packages in the freecad part.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -239,6 +252,7 @@ parts:
       - openscad  # OpenSCAD
       - povray # Raytracing
       - povray-includes # Raytracing
+      - openscad
```
Comment (luzpaz) [inline snap/snapcraft.yaml:255]: although, openscad is already included  3 lines up. So is this line necessary ?

Comment (ScarlettGatelyMoore) [reply snap/snapcraft.yaml:255]: It is not! I am surprised snapcraft didn't fail. New PR to remove.
https://github.com/FreeCAD/FreeCAD-snap/pull/94

Comment (chennes) [reply snap/snapcraft.yaml:255]: Thanks: we can drop those povray installs as well, Raytracing is gone.

**Pull request merged**